File: 0342-power-of-four/0342-power-of-four.cpp
class Solution {
public:
    bool isPowerOfFour(int n) {
        if(n==1) return 1;
        if(n%4!=0 || n<1) return 0;
        while(n>1)
        {
            if(n%4==0) n/=4;
            else return 0;
        }
        return 1;
    }
};

File: 0342-power-of-four/0342-power-of-four.py
class Solution(object):
    def isPowerOfFour(self, n):
        if n==1:
            return True
        if n<1 or n%4!=0:
            return False
         
        while n>1:
            if n%4!=0:
                return False
            n/=4

        return True
        """
        :type n: int
        :rtype: bool
        """
        